rsmtool/notebooks/explanations/shap_values.ipynb

```diff
@@ -0,0 +1,207 @@
+    "\n",
+    "- If you wish to analyze shap values manually, please refer to the `*shap*.csv` files in the `output` directory.\n",
+    "\n",
+    "- If you wish to use the generated shap explanation object for additional processing, you may unpickle the `<id>_explanation.pkl` file to get access to it. If you chose specific examples via `sample_range` or `sample_size`, you can find them in the `<id>_ids.pkl` file where they are stored as a mapping between the position of the example in the dataset and the ID of the example."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Shap values summary"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is a quick textual summary of your shap values. Please refer to the Plots section below for visualizations. All values are rounded to $10^{-3}$ unless specified otherwise."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Top 5 features by mean absolute Shap value\n",
+    "\n",
+    "The following table shows the top 5 features in terms of mean absolute shap value, i.e., the top 5 features with the biggest mean impact on model predictions. Note that the table also includes the maximum and minimum absolute values for each feature. \n",
+    "\n",
+    "*Note: if your model has less than or equal to 5 features, all of them will be displayed.*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "if len(df_abs_shap) > 5:\n",
+    "    top_5_table = HTML(df_abs_shap[0:5].to_html(classes=['sortable'],\n",
+    "                                                float_format=float_format_func))\n",
+    "else:\n",
+    "    top_5_table = HTML(df_abs_shap.to_html(classes=['sortable'],\n",
+    "                                           float_format=float_format_func))\n",
+    "\n",
+    "display(top_5_table)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "mean_values = df_abs_shap[\"abs. mean shap\"]\n",
+    "rows_with_zero_mean_values = df_abs_shap[df_abs_shap[\"abs. mean shap\"] == 0]\n",
+    "\n",
+    "msg = (\"#### Features with zero mean Shap value\\n The featuers in table below \"\n",
+    "       \"likely did not contribute to the model decisions. Assuming the examples passed \"\n",
+    "       \"were sufficient in number and representative of the data the model usually \"\n",
+    "       \"encounters, the features in this table are not useful to the model.\\n \"\n",
+    "       \"**IMPORTANT**: Before you draw this conclusion, please make sure that those \"\n",
+    "       \"features were not simply 0 in all the examples you provided the model.\")\n",
```

Comment:
`in all the examples you provided the model` - does that mean in the training data?

Reply:
Also, maybe this section should come after the `Bottom 5 features by mean absolute Shap value`?

Reply:
Yes, I think it means training data. Although I suppose it was all zero in the background or explainable data, it would also not come up as very important?

Reply:
I think it may not be the training data.

> the value for a feature indicates, for every model decision passed to the explainer, how the model decision would be impacted by removing that feature.

possibly shap removes the feature by setting it to 0?

Reply:
I don't think it would do it that way because then how would it distinguish the case where a feature was actually zero in all the instances?

Reply:
it would add a comment like the above...

Reply:
Features that have a 0 mean shap value did not contribute to the model decisions in all rows that were passed. If you passed a representative sample, then that means that this feature is not useful to your model. 

However, if the sample is not representative, this could be misleading:
So imagine you train a model on 10k rows. You pass 20 rows to a SHAP explanation. Coincidentally, in those 20 rows a feature that usually has an impact does not have an impact for those 20 decisions, for whichever reason. That's why that disclaimer is there.

So this is not about training data for the model, but about the data you pass to SHAP.

hope that helps

Reply:
Thanks @remo-help  I think that makes sense. So, the disclaimer is about whether the sample you are passing to SHAP to explain is representative of your real training data or not. If the real training data also had that feature as always zero, your representative explain data will have it too. It's only about when the two distributions don't match up. I will clarify.

Reply:
The real training data might have actual values for that feature, but your model may have decided to ignore that feature altogether. That would also get you a mean SHAP value of 0. This is pretty unlikely of course, but in theory possible.

Reply:
Yeah that would be pretty unlikely. In any case, I will make the disclaimer clearer.